### ISSUE TITLE
Pip unable to get the version of the source tarball of 5.3.1

### ISSUE THREAD

**Issue by aptalca** [pikepdf/pikepdf#367]
Sorry to bug you with another potentially pip/pypi issue but wanted to make sure it was a problem with them before I went to them with it (not sure where pip tries to get the version info from).

When I try to install version 5.3.1 from source on arm, pip complains that the filename has 5.3.1 but the metadata has 0.0

Here's the log from me attempting to build the wheel:
```
Collecting pikepdf

  Downloading pikepdf-5.3.1.tar.gz (2.4 MB)

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.4/2.4 MB 10.9 MB/s eta 0:00:00

  Installing build dependencies: started

  Running command pip subprocess to install build dependencies

  Looking in links: https://wheel-index.linuxserver.io/ubuntu/

  Collecting setuptools>=52

    Using cached setuptools-63.1.0-py3-none-any.whl (1.2 MB)

  Collecting setuptools-scm[toml]>=7.0.3

    Downloading setuptools_scm-7.0.4-py3-none-any.whl (42 kB)

       ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.9/42.9 kB 2.3 MB/s eta 0:00:00

  Collecting wheel>=0.35

    Using cached wheel-0.37.1-py2.py3-none-any.whl (35 kB)

  Collecting pybind11>=2.9.1

    Downloading pybind11-2.9.2-py2.py3-none-any.whl (213 kB)

       ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 213.6/213.6 kB 6.0 MB/s eta 0:00:00

  Collecting packaging>=20.0

    Downloading packaging-21.3-py3-none-any.whl (40 kB)

       ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.8/40.8 kB 4.3 MB/s eta 0:00:00

  Collecting typing-extensions

    Downloading typing_extensions-4.3.0-py3-none-any.whl (25 kB)

  Collecting tomli>=1.0.0

    Downloading tomli-2.0.1-py3-none-any.whl (12 kB)

  Collecting pyparsing!=3.0.5,>=2.0.2

    Downloading pyparsing-3.0.9-py3-none-any.whl (98 kB)

       ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 98.3/98.3 kB 8.2 MB/s eta 0:00:00

  Installing collected packages: wheel, typing-extensions, tomli, setuptools, pyparsing, pybind11, packaging, setuptools-scm

  Successfully installed packaging-21.3 pybind11-2.9.2 pyparsing-3.0.9 setuptools-63.1.0 setuptools-scm-7.0.4 tomli-2.0.1 typing-extensions-4.3.0 wheel-0.37.1

  Installing build dependencies: finished with status 'done'

  Getting requirements to build wheel: started

  Running command Getting requirements to build wheel

  running egg_info

  writing src/pikepdf.egg-info/PKG-INFO

  writing dependency_links to src/pikepdf.egg-info/dependency_links.txt

  writing requirements to src/pikepdf.egg-info/requires.txt

  writing top-level names to src/pikepdf.egg-info/top_level.txt

  reading manifest file 'src/pikepdf.egg-info/SOURCES.txt'

  adding license file 'LICENSE.txt'

  adding license file 'licenses/license.wheel.txt'

  writing manifest file 'src/pikepdf.egg-info/SOURCES.txt'

  /tmp/pip-build-env-5a_5hngr/overlay/lib/python3.8/site-packages/setuptools/config/setupcfg.py:463: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.

    warnings.warn(msg, warning_class)

  /tmp/pip-build-env-5a_5hngr/overlay/lib/python3.8/site-packages/setuptools_scm/git.py:284: UserWarning: git archive did not support describe output

    warnings.warn("git archive did not support describe output")

  Getting requirements to build wheel: finished with status 'done'

  Preparing metadata (pyproject.toml): started

  Running command Preparing metadata (pyproject.toml)

  running dist_info

  creating /tmp/pip-modern-metadata-9edcwxhy/pikepdf.egg-info

  writing /tmp/pip-modern-metadata-9edcwxhy/pikepdf.egg-info/PKG-INFO

  writing dependency_links to /tmp/pip-modern-metadata-9edcwxhy/pikepdf.egg-info/dependency_links.txt

  writing requirements to /tmp/pip-modern-metadata-9edcwxhy/pikepdf.egg-info/requires.txt

  writing top-level names to /tmp/pip-modern-metadata-9edcwxhy/pikepdf.egg-info/top_level.txt

  writing manifest file '/tmp/pip-modern-metadata-9edcwxhy/pikepdf.egg-info/SOURCES.txt'

  reading manifest file '/tmp/pip-modern-metadata-9edcwxhy/pikepdf.egg-info/SOURCES.txt'

  adding license file 'LICENSE.txt'

  adding license file 'licenses/license.wheel.txt'

  writing manifest file '/tmp/pip-modern-metadata-9edcwxhy/pikepdf.egg-info/SOURCES.txt'

  creating '/tmp/pip-modern-metadata-9edcwxhy/pikepdf-0.0.dist-info'

  adding license file "licenses/license.wheel.txt" (matched pattern "licenses/license.wheel.txt")

  /tmp/pip-build-env-5a_5hngr/overlay/lib/python3.8/site-packages/setuptools_scm/git.py:284: UserWarning: git archive did not support describe output

    warnings.warn("git archive did not support describe output")

  /tmp/pip-build-env-5a_5hngr/overlay/lib/python3.8/site-packages/setuptools/config/setupcfg.py:463: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.

    warnings.warn(msg, warning_class)

  Preparing metadata (pyproject.toml): finished with status 'done'

Discarding https://files.pythonhosted.org/packages/6e/e2/3b2a8c2d558ccb908be16efee2337a56f3f7b2a9810f090ae1d8b4162af1/pikepdf-5.3.1.tar.gz#sha256=b8c9e5cc1c73fa4d0dae9f4b184adbc72949973a011f2746c851affec3c0ede3 (from https://pypi.org/simple/pikepdf/) (requires-python:>=3.7): Requested pikepdf from https://files.pythonhosted.org/packages/6e/e2/3b2a8c2d558ccb908be16efee2337a56f3f7b2a9810f090ae1d8b4162af1/pikepdf-5.3.1.tar.gz#sha256=b8c9e5cc1c73fa4d0dae9f4b184adbc72949973a011f2746c851affec3c0ede3 has inconsistent version: filename has '5.3.1', but metadata has '0.0'
```

Then it moves on to 5.3.0 and attempts to build that.

Thanks

EDIT: This is with the latest pip, setuptools and wheel. Same behavior is observed on both python 3.8 and 3.10 on ubuntu focal and jammy, respectively.

**Comment by jbarlow83** [comment]
Reported as https://github.com/pypa/setuptools_scm/issues/742

You can force the correct version with `env SETUPTOOLS_SCM_PRETEND_VERSION=5.3.1 pip install pikepdf.tar.gz` or use a wheel if available. (Since there are aarch64 linux wheels now.)

**Comment by aminalaee** [comment]
@jbarlow83 Is there a limitation to why Apple Silicon Python 3.8 is not added?

**Comment by jbarlow83** [comment]
Fixed for 5.3.2
